crates/context/config/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+#![allow(single_use_lifetimes)]
+
+use std::borrow::Cow;
+use std::time;
+
+use serde::{Deserialize, Serialize};
+
+pub mod repr;
+pub mod types;
+
+use repr::Repr;
+use types::{Application, Capability, ContextId, ContextIdentity, SignerId};
+
+pub type Timestamp = u64;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+pub struct Request<'a> {
+    #[serde(borrow, flatten)]
+    pub kind: RequestKind<'a>,
+
+    pub signer_id: Repr<SignerId>,
+    pub timestamp_ms: Timestamp,
+}
+
+impl<'a> Request<'a> {
+    pub fn new(signer_id: SignerId, kind: RequestKind<'a>) -> Self {
+        let timestamp_ms = time::SystemTime::now()
+            .duration_since(time::UNIX_EPOCH)
+            .expect("system time is before epoch?")
```

Comment:
Technically this isn't the only reason it could be rejected.

Reply:
why else would it be rejected?

Reply:
It could overflow.

Reply:
Hm, this isn't `chrono` btw, this is stdlib and it doesn't seem to

https://github.com/rust-lang/rust/blob/c2f74c3f928aeb503f15b4e9ef5778e77f3058b8/library/std/src/time.rs#L540-L542

https://github.com/rust-lang/rust/blob/c2f74c3f928aeb503f15b4e9ef5778e77f3058b8/library/std/src/sys/pal/unix/time.rs#L45-L47

https://github.com/rust-lang/rust/blob/c2f74c3f928aeb503f15b4e9ef5778e77f3058b8/library/std/src/sys/pal/unix/time.rs#L135-L166